lib/index.js

```diff
@@ -0,0 +1,195 @@
+'use strict';
+const yaml = require('js-yaml');
+
+/**
+ * This library is responsible for generating an engine spec compatible with Docker Engine v1.36
+ * from a docker-compose formatted YAML file (version 3).
+ */
+
+function isNumeric(value) {
+  return !isNaN(parseFloat(value)) && isFinite(value);
+}
+
+function dockerComposeToApi(input) {
+  if (typeof input === "string") {
+    input = yaml.safeLoad(input);
+  }
+
+  const name = Object.keys(input)[0];
+  let containerName = name;
+  if ('container_name' in input[name]) {
+    containerName = input[name]['container_name'];
+  }
+
+  let spec = {
+    Name: containerName,
+    TaskTemplate: {
+      ContainerSpec: {
+        Image: input[name]['image']
+      }
+    }
+  };
+
+  if ('command' in input[name]) {
+    if (Array.isArray(input[name]['command'])) {
+      spec.TaskTemplate.ContainerSpec['Command'] = input[name]['command'];
+    } else {
+      spec.TaskTemplate.ContainerSpec['Command'] = [ input[name]['command'] ];
+    }
+  }
+
+  if ('environment' in input[name]) {
+    spec.TaskTemplate.ContainerSpec['Env'] = [];
+
+    // Convert { test: 1 } to 'test=1' format.
+    for (var key in input[name]['environment']) {
+      spec.TaskTemplate.ContainerSpec['Env'].push(key + "=" + input[name]['environment'][key]);
+    }
+  }
+
+  if ('labels' in input[name]) {
+    spec.TaskTemplate.ContainerSpec['Labels'] = {};
+    // Labels can come in both array and dictionary format.
+    if (isNumeric(Object.keys(input[name]['labels'])[0])) {
+      // Array format
+      for (var i = 0; i < input[name]['labels'].length; i++) {
+        const parts = input[name]['labels'][i].split("=");
+        if (parts.length == 2) {
+          spec.TaskTemplate.ContainerSpec['Labels'][parts[0]] = parts[1];
+        }
+      }
+    } else {
+      // Dictionary format
+      spec.TaskTemplate.ContainerSpec['Labels'] = input[name]['labels'];
+    }
+  }
+
+  if ('configs' in input[name]) {
+    spec.TaskTemplate.ContainerSpec['Configs'] = [];
+    for (var key in input[name]['configs']) {
+      if ('file' in input[name]['configs'][key]) {
+        spec.TaskTemplate.ContainerSpec['Configs'].push({
+          ConfigID: key,
+          ConfigName: key,
+          File: input[name]['configs'][key].file
+        });
+      }
+    }
+  }
+
+  if ('secrets' in input[name]) {
+    spec.TaskTemplate.ContainerSpec['Secrets'] = [];
+    for (var key in input[name]['secrets']) {
+      if ('file' in input[name]['secrets'][key]) {
+        spec.TaskTemplate.ContainerSpec['Secrets'].push({
+          SecretID: key,
+          SecretName: key,
+          File: input[name]['secrets'][key].file
+        });
+      }
+    }
+  }
+
+  if ('volumes' in input[name] && input[name]['volumes'].length > 0) {
```

Comment:
wonder if it would be cleaner to put all of these in a `./lib/volumes.js` file that way the file doesn't just keep growing.  then you could loop over the object and just call require(`./lib/${key}`)(obj)

Reply:
and throw an error if something is there that isn't supported